### PR TITLE
sqlite extension loading tests

### PR DESCRIPTION
**Pull request by alamminsalo**
Tests loading with/without calling `enable_load_extension()` call and calling extension function with result assert.

Added `libspatialite5` as test dependency.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -89,6 +90,7 @@ branches:
   only:
     - master
     - ಠ_ಠ
+    - sg-enable-load-extension
```
Comment (weiznich) [inline .travis.yml]: You should not change this branches.

File: diesel/src/sqlite/connection/raw.rs [hunk 2]
```diff
@@ -106,8 +106,8 @@ impl RawConnection {
         }
     }
 
-    pub fn enable_load_extension(&self) {
-        unsafe { ffi::enable_load_extension(self.internal_connection.as_ptr(), true) }
+    pub fn enable_load_extension(&self) -> i32 {
```
Comment (weiznich) [inline diesel/src/sqlite/connection/raw.rs]: Why do we return a `i32` here?

Comment (alamminsalo) [reply diesel/src/sqlite/connection/raw.rs]: changed to return QueryResult instead

File: .travis.yml (hunk 1 above)
Comment (alamminsalo) [reply .travis.yml]: removed this line

**Comment by alamminsalo** [comment]
What is the status of this pull? The requested changes were added

**Comment by weiznich** [comment]
This PR is opened against a old stale branch.

**Pull request closed without merging**